### PR TITLE
Add packaging workflow

### LINKED ISSUE #25: Mojopkg GH action
I could create some GH actions to auto run the tests and make mojopkg files for release :) I already created the script for another project and it's easy to adapt and low maintenance

### PR DESCRIPTION
**Pull request by Moosems**
Fixes #25

### CODE REVIEW

**Comment by Moosems** [comment]
You will need to create a secret key for it that allows it to create releases and copy the code from the repository with the name `TOKEN`

File: .github/workflows/package.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Package and Release
+
+on:
+  push:
+    branches:
+      - main
+
```
Comment (saviorand) [inline .github/workflows/package.yml:7]:
```suggestion
permissions:
  contents: write
```

File: .github/workflows/package.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Add to PATH
+        run: echo "/home/runner/.modular/pkg/packages.modular.com_mojo/bin" >> $GITHUB_PATH
+
+      - name: Create package
+        run: mojo package lightbug_http -o lightbug_http.mojopkg
+
+      - name: Upload package to release
+        uses: svenstaro/upload-release-action@v2
+        with:
+          repo_token: ${{ secrets.TOKEN }}
```
Comment (saviorand) [inline .github/workflows/package.yml:34]:
```suggestion

```

**Comment by saviorand** [comment]
@Moosems just pushed a commit to main that should fix the packaging issue you were having.
See a couple small edits below: basically I think we can just use the default `GITHUB_TOKEN` for the release job with write permissions, no need for a custom token. Tested this on a private copy of the main repo and it works.
Can you explain why the `modular auth examples` works for authenticating with modular? 😄 this is very surprising to me. 
Otherwise, with these small changes should be good to merge

**Comment by saviorand** [comment]
I assume it uses some existing token. I could also pass my own token as an env variable

**Comment by saviorand** [comment]
Thanks a lot for the contribution! Merged

**Pull request merged**